pkg/router/router.go

```diff
@@ -18,25 +19,30 @@ func New(productHandler *handlers.ProductsHandler) *mux.Router {
 
 	// Get Router
 	getRouter := router.Methods(http.MethodGet).Subrouter()
+	getRouter.Use(tokenValidation.Middleware)
 	getRouter.HandleFunc("/products", productHandler.GetProducts)
 	getRouter.HandleFunc("/products/{id:[0-9a-z-]+}", productHandler.GetProductByID)
 
 	//Health Check
-	getRouter.HandleFunc("/health/live", productHandler.LivenessCheck)
-	getRouter.HandleFunc("/health/ready", productHandler.ReadinessCheck)
+	healthRouter := router.Methods(http.MethodGet).Subrouter()
```

Comment:
Do you know if setting another http.Method get router stop the other get endpoints from being exposed